app/serializers/study_group_serializer.rb

```diff
@@ -0,0 +1,7 @@
+class StudyGroupSerializer
+  include FastJsonapi::ObjectSerializer
+
+  attributes :title, :description
+  belongs_to :user, if: Proc.new { |record, params| record.association(:user).loaded? }
+  belongs_to :course, if: Proc.new { |record, params| record.association(:course).loaded? }
```

Comment:
Di base il serializer:
1) prende il model 
2) per ogni relazione che questo model possiede, aggiunge alle "relationships", type e ID dei model relazionati. 

Questo consuma query sul DB, perché l’ID di ogni model relazionato, in qualche modo lo deve pescare.

Aggiungendo questo ```if:``` ci assicuriamo che le informazioni sulle relazioni le aggiunga solamente se queste sono già state pescate dal DB con ```Model.includes(:altro_model)```.

Al momento non vedo uno scenario in cui nelle response ci serva avere solo gli ID (senza gli altri campi) dei model relazionati a quello di interesse, perciò così facendo risparmiamo delle query.

Reply:
capito, quindi per correttezza dovremmo rifare anche gli altri serializer